core/src/main/scala/cats/data/NonEmptyMapImpl.scala

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2018 Luka Jacobowitz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats
+package data
+
+import cats.instances.sortedMap._
+import cats.kernel._
+import cats.{Always, Apply, Eval, Foldable, Functor, Later, NonEmptyTraverse, Now, SemigroupK, Show}
+
+import scala.collection.immutable._
+
+private[data] object NonEmptyMapImpl extends NonEmptyMapInstances with Newtype2 {
+
+  private[cats] def create[K, A](m: SortedMap[K, A]): Type[K, A] =
+    m.asInstanceOf[Type[K, A]]
+
+  private[cats] def unwrap[K, A](m: Type[K, A]): SortedMap[K, A] =
+    m.asInstanceOf[SortedMap[K, A]]
+
+  def fromMap[K: Order, A](as: SortedMap[K, A]): Option[NonEmptyMap[K, A]] =
+    if (as.nonEmpty) Option(create(as)) else None
+
+  def fromMapUnsafe[K: Order, A](m: SortedMap[K, A]): NonEmptyMap[K, A] =
+    if (m.nonEmpty) create(m)
+    else throw new IllegalArgumentException("Cannot create NonEmptyMap from empty map")
+
+  def apply[K: Order, A](head: (K, A), tail: SortedMap[K, A]): NonEmptyMap[K, A] =
+    create(SortedMap(head)(Order[K].toOrdering) ++ tail)
```

Comment:
This and I think some other methods don't follow the convention that we kind of settled on in the early days of Cats [here](https://github.com/typelevel/cats/issues/27). Admittedly, that never got added to the guidelines, and I'm not sure whether we have been consistent about that in other places.

Reply:
I personally don't have a strong opinion on this and didn't know we had such a convention :smile:, you're right we should maybe reopen discussion and add it to the guideline, but for now I'll just change it, so we can avoid calling `apply` on the companion :+1: Thanks!